thunder/torch/__init__.py

```diff
@@ -5331,6 +5336,223 @@ def wait(slf) -> None:
+    ex.register_implementation(sym, op, checker=_always_executable)
+
+    from thunder.core.transforms import augmented_forward_impls, backward_impls
+
+    augmented_forward_impls[sym.id] = augmented_forward_adaptor(op)
+
+    def _vjp_impl_wrapper(residules, *gs):
+        return _vjp_impl(torchfn, residules, *gs)
+
+    bwd_op = ex.register_operator(torchfn.__name__ + "_vjp", meta=backward_adaptor(torchfn), fn=_vjp_impl_wrapper)
+    ex.register_implementation(bwd_op.id, bwd_op, checker=_always_executable)
+    backward_impls[sym.id] = bwd_op
+
+
+def _get_fake_arg(inp: Any, fake_mode):
+    if inp is None:
+        return inp
+    if isinstance(inp, NumberProxy):
+        if inp.value == None:
+            raise NotImplementedError("Unsupported for NumberProxy.value=None")
+        else:
+            return inp.value
+    elif isinstance(inp, TensorProxy):
+        return fake_mode.from_tensor(
+            torch.empty(
+                inp.shape,
+                dtype=_thunder_to_torch_dtype_map[inp.dtype],
+                requires_grad=inp.requires_grad,
+                device=devices.to_torch_device(inp.device),
+            )
+        )
+    elif isinstance(inp, (TupleProxy, ListProxy, DictProxy)):
+        return inp._value
+    elif isinstance(inp, (torch.dtype, torch.device, torch.Size, torch.memory_format, str, bool, int, float, complex)):
+        return inp
+    elif isinstance(inp, devices.Device):
+        return devices.to_torch_device(inp)
+    elif isinstance(inp, dtypes.dtype):
+        return to_torch_dtype(inp)
+    else:
+        raise NotImplementedError(f"Unsupported type: {builtins.type(inp)}")
+
+
+def _fake_type_to_thunder(inp: Any):
+    from thunder.core.proxies import _cls_to_number_proxy_map
+    from torch._subclasses.fake_tensor import FakeTensor
+
+    if inp is None:
+        return inp
+    if isinstance(inp, tuple(_cls_to_number_proxy_map.keys())):
+        return numberproxy(builtins.type(inp), inp)
+    elif isinstance(inp, FakeTensor):
+        return TensorProxy(
+            shape=inp.shape,
+            device=to_device(inp.device),
+            dtype=_torch_to_thunder_dtype_map[inp.dtype],
+            requires_grad=inp.requires_grad,
+        )
+    elif isinstance(inp, torch.Size):
+        return tuple(inp)
+    elif isinstance(inp, torch.device):
+        return to_device(inp)
+    elif isinstance(inp, torch.dtype):
+        return _torch_to_thunder_dtype_map[inp]
+    elif isinstance(inp, (str, bool, int, float, complex)):
+        return inp
+    else:
+        raise NotImplementedError(f"Unsupported type: {builtins.type(inp)}")
+
+
+def augmented_forward_adaptor(sym_op: Callable):
+    def wrapper(*args, **kwargs):
+        from thunder.core.transforms import VJPDual
+
+        out = sym_op(*args, **kwargs)
+        primal = out if isinstance(out, tuple) else (out,)
+        residuals = ((args, kwargs),)
+        return VJPDual(primal, residuals)
```

Comment:
The function signature of `wrapper` in `backward_adaptor` uses the `saved_for_backward` name for this variable. Let's use the same variable name across the functions for readability.
```suggestion
        saved_for_backward = ((args, kwargs),)
        return VJPDual(primal, saved_for_backward)
```